framework/wazuh/agent.py

```diff
@@ -627,12 +627,12 @@ def get_agent_groups(group_list: list = None, offset: int = 0, limit: int = None
 
                 affected_groups.append(group)
 
-        data = process_array(affected_groups, offset=offset, limit=limit, allowed_sort_fields=GROUP_FIELDS,
+        data = process_array(affected_groups, allowed_sort_fields=GROUP_FIELDS,
                             sort_by=sort_by, sort_ascending=sort_ascending, search_text=search_text,
                             complementary_search=complementary_search, q=q, allowed_select_fields=GROUP_FIELDS,
                             select=select, distinct=distinct, required_fields=GROUP_REQUIRED_FIELDS)
         result.affected_items = data['items']
-        result.total_affected_items = data['totalItems']
+        result.total_affected_items = query_data['totalItems']
```

Comment:
I just saw that there are some failing checks and it seems those are related with this line. If no filters are applied in the process_array, the number of total items should be `query_data['totalItems']` (all items in the DB). But if any filter is applied (for exmaple, `q=name=test_group_3`), the number of total_items should be 1.

This should do the trick, but review the unit and API tests to make sure it works as expected:
```suggestion
        result.total_affected_items = query_data['totalItems'] if limit == data['totalItems'] else data['totalItems']
```